src/Select.js

```diff
@@ -494,23 +497,42 @@ var Select = React.createClass({
 		this._optionsFilterString = event.target.value;
 
 		if (this.props.onInputChange) {
-			this.props.onInputChange(event.target.value);
+			this.props.onInputChange(this._optionsFilterString);
 		}
 
 		if (this.props.asyncOptions) {
+			var callAsyncLoad = function() {
+				this.loadAsyncOptions(this._optionsFilterString, {
```

Comment:
Quick note that `this._optionsFilterString` is preferable to `event.target.value`, since in the case of this function it doesn't require the argument to be redefined in its closure.